File: programmers/2023-02-03-2xn타일링.js
const solution = (n) => {
    const dp = [0, 1, 2];
    
    for (let i = 3; i <= n; i++) {
        dp[i] = (dp[i-1] + dp[i-2]) % 1000000007
    };

    return dp[n];
};

console.log(solution(4));

File: programmers/2023-02-03-2xn타일링.py
def solution(n):
    dp = [0] * n
    
    if n < 3:
        return n
    
    dp[0] = 1
    dp[1] = 2
    
    for i in range(2, n):
        dp[i] = dp[i-2] + dp[i-1]
    
    return dp[n-1]

print(solution(5))